mbotmake2/grammars/thumbnails.py

```diff
@@ -4,19 +4,26 @@
 
 grammar = Grammar(
     r"""
-Document = Timestamp Thumbnail+
-Thumbnail = BlankLine+ Header Payload+ Footer
+Document = (Timestamp / HeaderBlock) Thumbnail+
+Thumbnail = Header Payload+ Footer
 
 BlankLine = ~r"[ \t\r]*\n"i
-Timestamp = "; generated by " ~r"[^\n]+"i newline
+Timestamp = "; generated by " ~r"[^\n]+" newline+
 
-Header = ";\n; thumbnail begin " Integer "x" Integer " " Integer newline
+HeaderBlock = HeaderBeginTag (!HeaderEndTag Comment)+ HeaderEndTag newline+
+HeaderBeginTag = "; HEADER_BLOCK_START\n"
+HeaderEndTag = "; HEADER_BLOCK_END\n"
```

Comment:
In plain English: a GCode file header block, generated by OrcaSlicer, is any lines sandwiched between `HEADER_BLOCK_START` and `HEADER_BLOCK_END`.